app/ide-desktop/lib/client/src/authentication.ts

```diff
@@ -129,24 +123,30 @@ function initIpc() {
  *
  * All URLs that aren't deep links (i.e., URLs that don't use the {@link common.DEEP_LINK_SCHEME}
  * protocol) will be ignored by this handler. Non-deep link URLs will be handled by Electron. */
-function initOpenUrlListener(window: () => electron.BrowserWindow) {
+function initOpenUrlListener(getLastFocusedWindow: () => electron.BrowserWindow | null) {
     urlAssociations.registerUrlCallback(url => {
-        onOpenUrl(url, window)
+        onOpenUrl(url, getLastFocusedWindow)
     })
 }
 
 /** Handle the 'open-url' event by parsing the received URL, checking if it is a deep link, and
  * sending it to the appropriate BrowserWindow via IPC.
  *
- * @param url - The URL to handle.
- * @param window - A function that returns the BrowserWindow to send the parsed URL to. */
-export function onOpenUrl(url: URL, window: () => electron.BrowserWindow) {
+ * @param url - The URL to handle. */
+export function onOpenUrl(url: URL, getLastFocusedWindow: () => electron.BrowserWindow | null) {
     logger.log(`Received 'open-url' event for '${url.toString()}'.`)
     if (url.protocol !== `${common.DEEP_LINK_SCHEME}:`) {
         logger.error(`'${url.toString()}' is not a deep link, ignoring.`)
     } else {
-        logger.log(`'${url.toString()}' is a deep link, sending to renderer.`)
-        window().webContents.send(ipc.Channel.openDeepLink, url.toString())
+        const window = getLastFocusedWindow()
+        if (window == null) {
+            logger.error(
+                `'${url.toString()}'` + ' is a deep link but there is no renderer to send it to.'
+            )
```

Comment:
In the previous version of the code, this branch was impossible. Did it change? In what circumstances `getLastFocusedWindow` returns `null`?

Reply:
in the previous version this branch wasn't impossible, just suppressed by a non-null assertion (`() => this.window!`). i'd imagine even in the old version it *may* have happened if `args.options.window` was `false`. i've checked on this branch and can confirm that `enso.AppImage enso://authentication/register` does trigger this branch, can test on develop soon to check whether this is a regression.

Reply:
hm. just tested on develop... I get the `We are the first instance of the application. This is not expected.`, because of course, on current develop the only *supported* usecase for deep links is for sign in/sign out redirects